presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceStreamReader.java

```diff
@@ -41,16 +46,18 @@
 public class SliceStreamReader
         implements StreamReader
 {
+    private static final int INSTANCE_SIZE = ClassLayout.parseClass(SliceStreamReader.class).instanceSize();
+
     private final StreamDescriptor streamDescriptor;
     private final SliceDirectStreamReader directReader;
     private final SliceDictionaryStreamReader dictionaryReader;
     private StreamReader currentReader;
 
-    public SliceStreamReader(StreamDescriptor streamDescriptor)
+    public SliceStreamReader(StreamDescriptor streamDescriptor, AggregatedMemoryContext systemMemoryContext)
     {
         this.streamDescriptor = requireNonNull(streamDescriptor, "stream is null");
-        directReader = new SliceDirectStreamReader(streamDescriptor);
-        dictionaryReader = new SliceDictionaryStreamReader(streamDescriptor);
+        directReader = new SliceDirectStreamReader(streamDescriptor, systemMemoryContext.newLocalMemoryContext());
```

Comment:
These won't compile (a new change was merged to master recently, rebase please), you need to pass a `String` tag: `newLocalMemoryContext(SliceStreamReader.class.getSimpleName())`.

Let's discuss offline as these will require additional changes to this PR (around `StreamReaders::createStreamReader`).